docs/content/documentation/query.md

```diff
@@ -61,7 +61,8 @@ retrieved by repositioning the cursor.
 >
 > * The `Statement` must be created with a `ResultSet` type of `ResultSet.TYPE_FORWARD_ONLY`. This is the default,
 > so no code will need to be rewritten to take advantage of this, but it also means that you cannot scroll backwards or 
-> otherwise jump around in the `ResultSet`.
+> otherwise jump around in the `ResultSet`. Don't use `ResultSet.HOLD_CURSORS_OVER_COMMIT` since cursors holdability is
```

Comment:
Not sure I'd suggest not using it, however it is useful to document that it will retrieve all the results.

Reply:
On the client side you choose to use a cursor based `ResultSet` to avoid loading all the results in memory.
This option is defeating its purpose since the JVM heap is filled with `org.postgresql.core.Tuple` until OOM.

Reply:
You can actually use a cursor based result set if you do it inside a transaction and set the fetch size.

Reply:
Being in a transaction is a listed requirement for having a cursor based result set,
it works fine until you add the `ResultSet.HOLD_CURSORS_OVER_COMMIT` option. 
Check the `((PgResultSet) rs).getLastUsedFetchSize()` value after the `executeQuery`:
- without cursor holdability option it returns the fetch size
- with cursor holdability option it returns the total rows, not the expected behavior for a cursor based result set.

Reply:
Seems like a bug rather than a documentation error. 
I trust you realize the only way to do this properly is to use WITH HOLD cursors and materialize the result set on the server ?

Reply:
My understanding is that PostgreSQL cannot maintain a cursor between transactions and this is simulated at the JDBC Driver level where all results are retrieved on the first call (in the first transaction).
I have no problem with this workaround but it should be documented that it prevents to get a true cursor based  result set, so it's expected to encounter OOM on large results.

Reply:
> My understanding is that PostgreSQL cannot maintain a cursor between transactions and this is simulated at the JDBC Driver level where all results are retrieved on the first call (in the first transaction). I have no problem with this workaround but it should be documented that it prevents to get a true cursor based result set, so it's expected to encounter OOM on large results.

Actually it is quite possible for PostgreSQL to maintain a cursor between transactions using a CURSOR WITH HOLD.

I agree the current behaviour should be documented. I don't agree that the docs should be telling people to not use the current behaviour.

Reply:
Yes, please rephrase it. Telling people not to use it is disingenuous.

Reply:
I tried, but indeed, it might be more appropriate to create an issue.

Reply:
I'd prefer we actually warned people of the potential to consume large amounts of memory. The current change doesn't really change much?